Services/Simulations.cs

```diff
@@ -214,7 +214,7 @@ public async Task<Models.Simulation> UpsertAsync(Models.Simulation simulation)
                 {
                     var msg = "Invalid ETag. Running simulation ETag is:'";
```

Comment:
Not related to your change but we should fix it. Append ETag to the message, otherwise log message is incomplete

Reply:
👍  fixed